packages/cli/src/options/beaconNodeOptions/network.ts

```diff
@@ -38,10 +43,59 @@ export type NetworkArgs = {
   "network.rateTrackerTimeoutMs"?: number;
 };
 
+function validateMultiaddrArg<T extends Record<string, string | undefined>>(args: T, key: keyof T): void {
+  if (args[key]) {
+    try {
+      multiaddr(args[key]);
+    } catch (e) {
+      throw new YargsError(`Invalid ${key as string}: ${(e as Error).message}`);
+    }
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function parseListenArgs(args: NetworkArgs) {
+  // If listenAddress is explicitly set, use it
+  // If listenAddress6 is not set, use defaultListenAddress
+  const listenAddress = args.listenAddress ?? (args.listenAddress6 ? undefined : defaultListenAddress);
+  const port = listenAddress ? args.port ?? defaultP2pPort : undefined;
+  const discoveryPort = listenAddress ? args.discoveryPort ?? args.port ?? defaultP2pPort : undefined;
+
+  // Only use listenAddress6 if it is explicitly set
+  const listenAddress6 = args.listenAddress6;
+  const port6 = listenAddress6 ? args.port6 ?? defaultP2pPort6 : undefined;
+  const discoveryPort6 = listenAddress6 ? args.discoveryPort6 ?? args.port6 ?? defaultP2pPort6 : undefined;
+
+  return {listenAddress, port, discoveryPort, listenAddress6, port6, discoveryPort6};
+}
+
 export function parseArgs(args: NetworkArgs): IBeaconNodeOptions["network"] {
-  const listenAddress = args.listenAddress || defaultListenAddress;
-  const udpPort = args.discoveryPort ?? args.port ?? defaultP2pPort;
-  const tcpPort = args.port ?? defaultP2pPort;
+  const {listenAddress, port, discoveryPort, listenAddress6, port6, discoveryPort6} = parseListenArgs(args);
+  // validate ip, ip6, ports
+  const muArgs = {
+    listenAddress: listenAddress ? `/ip4/${listenAddress}` : undefined,
+    port: listenAddress ? `/tcp/${port}` : undefined,
+    discoveryPort: listenAddress ? `/udp/${discoveryPort}` : undefined,
+    listenAddress6: listenAddress6 ? `/ip6/${listenAddress6}` : undefined,
+    port6: listenAddress6 ? `/tcp/${port6}` : undefined,
+    discoveryPort6: listenAddress6 ? `/udp/${discoveryPort6}` : undefined,
+  };
+
+  for (const key of [
+    "listenAddress",
+    "port",
+    "discoveryPort",
+    "listenAddress6",
+    "port6",
+    "discoveryPort6",
+  ] as (keyof typeof muArgs)[]) {
+    validateMultiaddrArg(muArgs, key);
```

Comment:
this allows us to fail fast if the user sets an incorrect ip address or port
eg: `--listenAddress ::` and `--listenAddress6 0.0.0.0` should error